### DISCUSSION TITLE
About the Key and communication encryption

### DISCUSSION THREAD

**Discussion by Deathleaves** [rustdesk/rustdesk#542] [Q&A] [answered]
Thank you for building such an excellent program!
I have deployed a server for self-use via Docker.  Following the manual (https://rustdesk.com/docs/en/self-host/synogy/), I can't find any file named "id_ed25519" to set the Key and encrypted connections. After setting "hbbs -k  " and  "hbbr -k  " , I can still connect to and remote PC without setting the Key. Is that because I used the demo version ( "hhbr -m demo")? If so how can I register my own account?
Can you update the manual for setting the Key? I think that is for a very old version. Thank you so much!

**Comment by rustdesk** [comment]
> Mount a host directory (e.g. Shared/test/) to /root, hbbs’s will generate some files in this directory

https://rustdesk.com/docs/en/self-host/synogy/images/mounted-dir.png?width=300px

Answer in the tutorial

**Reply by Deathleaves** [reply]
Thank you for the reply. I mounted a host directory (but with another name, docker/rustdesk) to /root before. The above files are still not generated after restarting hbbs/hbbr. Instead, I found the following files.
![image](https://user-images.githubusercontent.com/36256719/168741093-aa06c593-58e5-4061-8506-181dcfd8779c.png)

**Comment by rustdesk** [comment]
It is open source now, there is no `demo` etc any more.

**Reply by Deathleaves** [reply]
So I don't need to set "hbbs -m demo"/ "hbbr -m demo" as shown in the tutorial?
![image](https://user-images.githubusercontent.com/36256719/168741538-be40e563-abc7-41da-9225-ce71c0dd8c56.png)

**Reply by Deathleaves** [reply]
Without setting "-m demo", hbbs and hbbr are not initiated. How can I register my own email?
![image](https://user-images.githubusercontent.com/36256719/168746797-40bd1308-a219-4640-9c00-13ed557b35ff.png)

**Comment by rustdesk** [comment]
Update to new version please. `docker pull` or remove your old docker image.

**Reply by Deathleaves** [reply]
I am using the latest version.
![image](https://user-images.githubusercontent.com/36256719/168749880-9f51e6f3-7621-44c9-bd52-cc31d93a1213.png)

**Reply by Deathleaves** [reply]
Is 1.1.4 the latest version?
![image](https://user-images.githubusercontent.com/36256719/168750653-c686f7f7-603d-442f-9aab-a8ed1c265ae2.png)
After install hbbs/hbbr, do I still need to set "-m" to set my email address? If so, where can I register with Rustdesk? I can't find it on https://rustdesk.com/en/server/

**Reply by rustdesk** [reply]
https://github.com/rustdesk/rustdesk-server/releases/tag/1.1.5

**Reply by Deathleaves** [reply]
Oh, thanks. I pull it in the Docker of Synology .May the version in dockerhub is not updated. I will try the 1.1.5 version. I really appreciate your help.

**Comment by rustdesk** [comment]
Sorry, I do not know what caused your problem, I am ok to run with latest docker image without `-m` option.
![Screenshot 2022-05-17 at 3 10 36 PM](https://user-images.githubusercontent.com/71636191/168750922-a3442394-c7b4-4ecc-8c88-67e39566b312.png)

**Reply by Deathleaves** [reply]
I pulled the latetest rustdesk-server, but it still is the 1.1.4 version. Can you tell me how to install 1.1.5. I am using Synology. Thank you so much.
![image](https://user-images.githubusercontent.com/36256719/168800874-46b31e4a-a1c7-4220-b1b8-7563e809d493.png)

**Reply by rustdesk** [reply]
i am new to docker also

**Comment by rustdesk** [comment]
but，i do not know how to help u

**Reply by Deathleaves** [reply]
Thx, I will try some other ways

**Reply by Deathleaves** [reply]
I have found a way to solve this. Use digest to pull the latest image. In this way, however, the Tag is "none" and the pulled image is not found in Synology Docker UI. So use "docker tag " to give a tag with "latest". Then you can "docker run" or find it in Synology
![image](https://user-images.githubusercontent.com/36256719/168947678-00413dbc-4426-4ab8-9b9e-63fcbb760fe5.png)

**Reply by rustdesk** [reply]
No idea, why can not docker pull normally in Synogy? It is ok on my machines. Maybe the docker on Synogy is a buggy version.

**Reply by paspo** [reply]
I own a Synology and I can confirm: the docker management gui has a lot of "ui improvement potential".
When you use the wizard to create a container, it works ok; the issues arise when you want to update a container.

The best solution I came up is to add a **scheduled action** that I usually run manually, to be run as the **root user**, with this command:

```bash
docker run --rm -v /var/run/docker.sock:/var/run/docker.sock containrrr/watchtower --run-once containername1 containername2 containername3
```

If you omit the container names, it'll update all the containers.

**Comment by Deathleaves** [comment, marked as answer]
I have no idea what’s wrong with my Synology. Anyway, it is solved finally. Thx

**Comment by liuchengts** [comment]
The cause of the problem: the tag corresponding to latest on docker hub is not updated.

The latest version on docker hub so far should be: rustdesk-server:1.1.10-3

Solution: Specify the version to pull docker pull rustdesk-server:1.1.10-3
